### PR TITLE
proposed changes to issue prioritization.

### PR DESCRIPTION
**Pull request by shankarseal**
## Description

Proposes changes to issue prioritization policy.

## Testing

N/A.

## Documentation

Yes.

## Installation

N/A.

### CODE REVIEW

File: docs/Governance.md [hunk 1]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
 * Manage github settings such as branch protection rules and repository secrets.
 * Resolve any disputes among maintainers.
 
+## Project Milestones
+
+eBPF for Windows uses monthly milestones in the format `YYMM`. So, e.g. the milestone for June, 2024 will be called 2406.
+At any time there are four active milestones in the project. There will be an milestone for the current calendar month (`N`),
```
Comment (shpalani) [inline docs/Governance.md:122]: [nit]: Change "an milestone" --> "a milestone"

Comment (shpalani) [inline docs/Governance.md:122]: [nit]: Add a comma after "At any time,  "

File: docs/Governance.md [hunk 2]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
 * Manage github settings such as branch protection rules and repository secrets.
 * Resolve any disputes among maintainers.
 
+## Project Milestones
+
+eBPF for Windows uses monthly milestones in the format `YYMM`. So, e.g. the milestone for June, 2024 will be called 2406.
```
Comment (dthaler) [inline docs/Governance.md:121]:
```suggestion
eBPF for Windows uses monthly milestones in the format `YYMM`. So, e.g., the milestone for June, 2024 will be called 2406.
```

File: docs/Governance.md [hunk 3]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
+At any time there are four active milestones in the project. There will be an milestone for the current calendar month (`N`),
+and two others for the next two months (`N+1`), (`N+2`). And there is always a `Backlog` milestone for backlog issues to be
+addressed in the future. In the first triage meeting of a month, the previous milestone must be closed and a new `N+2` milestone must be created.
+
+During the weekly triage meetings, triage drivers MUST set an issue to one of the four milestones according to the priority of the issues.
```
Comment (dthaler) [inline docs/Governance.md:126]: Or leave it untriaged

File: docs/Governance.md [hunk 4]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
+
 ## Prioritizing Issues
 
-In the weekly triage meetings, new issues are assigned to collaborators and are set to a monthly milestone by which the issue is expected to be completed. It may not be possible for the assignees to address all the issues assigned to them by their target milestone. As a result issues may pile up in a given milestone. In such cases project admins can request that triage drivers and maintainers use *special labels* to prioritize issues as follows:
+To help prioritize issues, the triage drivers and maintainers MUST use *special labels* to prioritize issues as follows:
 ### P1:
-This label represents highest priority. This label should be used for bugs impacting users, new feature or enhancements that are urgently requested by users. Collaborators  should try to resolve these issues in the **current milestone**.
+This label represents highest priority. This label should be used for bugs impacting users, new feature or enhancements that are urgently
```
Comment (dthaler) [inline docs/Governance.md:132]:
```suggestion
This label represents highest priority. This label should be used for bugs impacting users, new features, or enhancements that are urgently
```

File: docs/Governance.md [hunk 5]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
+requested by users. `P1` issues MUST be set to the *current milestone (`N`)*. If there is not enough time left in the current milestone to address
+the `P1` issue, it must be set to the *next milestone (`N+1`)*.
 ### P2:
-This label is used for the next level of priority. Issues marked as P2 for the current milestone may be moved to the *next milestone* after discussing in the weekly triage meeting.
+This label is used for the next level of priority. Issues marked as `P2` can be set to milestones `N`, `N+1` (preferred) or `N+2`.
 ### P3:
-This label is used for low priority issues. Issues marked with this label may be moved to the *next milestone* or the *backlog* milestone, without needing further discussion in the weekly triage meeting.
+This label is used for low priority issues. Issues marked with `P3` label can be set to the milestones `N+1`, `N+2` or `Backlog`.
```
Comment (dthaler) [inline docs/Governance.md:138]:
```suggestion
This label is used for low priority issues. Issues marked with `P3` label can be set to the milestones `N+1`, `N+2`, or `Backlog`.
```

File: docs/Governance.md [hunk 6]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
+To help prioritize issues, the triage drivers and maintainers MUST use *special labels* to prioritize issues as follows:
 ### P1:
-This label represents highest priority. This label should be used for bugs impacting users, new feature or enhancements that are urgently requested by users. Collaborators  should try to resolve these issues in the **current milestone**.
+This label represents highest priority. This label should be used for bugs impacting users, new feature or enhancements that are urgently
+requested by users. `P1` issues MUST be set to the *current milestone (`N`)*. If there is not enough time left in the current milestone to address
+the `P1` issue, it must be set to the *next milestone (`N+1`)*.
 ### P2:
-This label is used for the next level of priority. Issues marked as P2 for the current milestone may be moved to the *next milestone* after discussing in the weekly triage meeting.
+This label is used for the next level of priority. Issues marked as `P2` can be set to milestones `N`, `N+1` (preferred) or `N+2`.
```
Comment (dthaler) [inline docs/Governance.md:136]: to confirm: no backlogging of P2?

File: docs/Governance.md [hunk 7]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
 ### P3:
-This label is used for low priority issues. Issues marked with this label may be moved to the *next milestone* or the *backlog* milestone, without needing further discussion in the weekly triage meeting.
+This label is used for low priority issues. Issues marked with `P3` label can be set to the milestones `N+1`, `N+2` or `Backlog`.
+`P3` issues must never be set to the current milestone (`N`).
```
Comment (dthaler) [inline docs/Governance.md:139]: I'm not sure I agree with this.  I think the project should be encouraging more contributors and allow first time contributors to take a "good first issue" and commit to it in the current milestone even if it's a P3.  I think the rule as stated will dissuade new contributors which I don't think would be healthy for the project.

File: docs/Governance.md [hunk 8]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
+
+In the last triage meeting of the monthly milestone, triage drivers or administrators must move all issues that do not have an active pull request
+to the next milestones. Ideally, all `P1` issues should be addressed in their designated milestones. However, if that does not happen for any reason,
```
Comment (dthaler) [inline docs/Governance.md:142]:
```suggestion
to the next milestone. Ideally, all `P1` issues should be addressed in their designated milestones. However, if that does not happen for any reason, a
```

File: docs/Governance.md (hunk 7 above)
Comment (dthaler) [reply docs/Governance.md:139]: I'd suggest "Issues marked with the `P3` label can be set to any milestone."

File: docs/Governance.md (hunk 6 above)
Comment (shankarseal) [reply docs/Governance.md:137]: Yes, only P3 can be backlogged.

File: docs/Governance.md [hunk 9]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
 ### P2:
-This label is used for the next level of priority. Issues marked as P2 for the current milestone may be moved to the *next milestone* after discussing in the weekly triage meeting.
+This label is used for the next level of priority. Issues marked as `P2` can be set to milestones `N`, `N+1` (preferred) or `N+2`.
 ### P3:
-This label is used for low priority issues. Issues marked with this label may be moved to the *next milestone* or the *backlog* milestone, without needing further discussion in the weekly triage meeting.
+This label is used for low priority issues. Issues marked with `P3` label can be set to any milestone including `Backlog`.
+
+In the last triage meeting of the monthly milestone, triage drivers or administrators must move all issues that do not have an active pull request
+to the next milestone. Ideally, all `P1` issues should be addressed in their designated milestones. However, if that does not happen for any reason,
+`P1` issue can be moved to `N+1` milestone. Similarly, overdue `P2` issues can be moved to milestones `N+1` or `N+2`. Overdue `P3` issues can be moved to
```
Comment (dthaler) [inline docs/Governance.md:143]:
```suggestion
a `P1` issue can be moved to `N+1` milestone. Similarly, overdue `P2` issues can be moved to milestones `N+1` or `N+2`. Overdue `P3` issues can be moved to
```
Grammar fix

File: docs/Governance.md [hunk 10]
```diff
@@ -116,15 +116,34 @@ Responsibilities:
...
+the `N+2` or `Backlog` milestones.
```
Comment (dthaler) [inline docs/Governance.md:144]:
```suggestion
the `N+1`, `N+2`, or `Backlog` milestones.
```
P3 can be set to any milestone.

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**